pallets/utxo/src/tests.rs

```diff
@@ -740,3 +681,814 @@ proptest! {
+    let test_fun = Box::new(
+        move |token_id,
+              alice_pub_key,
+              karl_pub_key,
+              token_utxo_hash,
+              token_utxo: TransactionOutput<H256>| {
+            Transaction {
+                inputs: vec![TransactionInput::new_empty(token_utxo_hash)],
+                outputs: vec![TransactionOutput::new_p2pk_with_data(
+                    0,
+                    H256::from(alice_pub_key),
+                    OutputData::TokenTransferV1 {
+                        token_id,
+                        amount: 1_000_000_001,
+                    },
+                )],
+                time_lock: Default::default(),
+            }
+            .sign_unchecked(&[token_utxo.clone()], 0, &karl_pub_key)
+        },
+    );
+    test_tx_issuance_for_transfer("output value must not exceed input value", test_fun);
+}
+
+#[test]
+fn test_token_transfer_exceed_amount_mlt() {
+    let test_fun = Box::new(
+        move |token_id: TokenId,
+              alice_pub_key,
+              karl_pub_key,
+              token_utxo_hash,
+              token_utxo: TransactionOutput<H256>| {
+            Transaction {
+                inputs: vec![TransactionInput::new_empty(token_utxo_hash)],
+                outputs: vec![TransactionOutput::new_p2pk_with_data(
+                    1_000_000_000,
+                    H256::from(alice_pub_key),
+                    OutputData::TokenTransferV1 {
+                        token_id: token_id.clone(),
+                        amount: 1_000_000_000,
+                    },
+                )],
+                time_lock: Default::default(),
+            }
+            .sign_unchecked(&[token_utxo.clone()], 0, &karl_pub_key)
+        },
+    );
+    test_tx_issuance_for_transfer("output value must not exceed input value", test_fun);
+}
+
+#[test]
+fn test_token_transfer_send_part_others_burn() {
+    let test_fun = Box::new(
+        move |token_id: TokenId,
+              alice_pub_key,
+              karl_pub_key,
+              token_utxo_hash,
+              token_utxo: TransactionOutput<H256>| {
+            Transaction {
+                inputs: vec![TransactionInput::new_empty(token_utxo_hash)],
+                outputs: vec![
+                    // Send only 30%, let's forget about another 70% of tokens
+                    TransactionOutput::new_p2pk_with_data(
+                        0,
+                        H256::from(alice_pub_key),
+                        OutputData::TokenTransferV1 {
+                            token_id: token_id.clone(),
+                            amount: 300_000_000,
+                        },
+                    ),
+                ],
+                time_lock: Default::default(),
+            }
+            .sign_unchecked(&[token_utxo.clone()], 0, &karl_pub_key)
+        },
+    );
+    test_tx_issuance_for_transfer("output value must not exceed input value", test_fun);
+}
+
+#[test]
+fn test_token_transfer() {
```

Comment:
Please also test multiple transfers of multiple kinds of tokens together. So issue TKN1, TKN2 and TKN3, with different amounts, and move them around, and make sure the math adds up.

Reply:
Done. Added `test_transfer_for_multiple_tokens`